### DISCUSSION TITLE
How do I get the selected text？

### DISCUSSION THREAD

**Discussion by lzxlzxlzxl** [ueberdosis/tiptap#3715] [General Questions]
Does anyone know how to get the selected text？

**Comment by leettaylor** [comment]
https://github.com/ueberdosis/tiptap/issues/369#issuecomment-942427603


```js
const { view, state } = editor
const { from, to } = view.state.selection
const text = state.doc.textBetween(from, to, '')
```

**Reply by Chaddeus** [reply]
Do you know if it is possible to capture the text to include a newline where there is a paragraph break? In my app, I need the text, but also need to know where each paragraph break is.

🙏

**Reply by deathg0d** [reply]
> Do you know if it is possible to capture the text to include a newline where there is a paragraph break? In my app, I need the text, but also need to know where each paragraph break is.
> 
> 🙏

Pretty simple:
```
const { view, state } = editor
const { from, to } = view.state.selection
const text = state.doc.textBetween(from, to, '\n')
```
